sdks/python/apache_beam/pvalue.py

```diff
@@ -141,7 +142,7 @@ def __or__(self, ptransform):
     return self.pipeline.apply(ptransform, self)
 
 
-class PCollection(PValue, Generic[T]):
+class PCollection(PValue, CompositeTypeHint):
```

Comment:
This should stay `Generic[T]`, as it provides the same `[]` functionality but is understood by tools like mypy and is familiar to Python users.

Reply:
Switched it back